### PR TITLE
feat: [CDS-78136]: Make barriers steps work within looping strategies

### PR DESCRIPTION
**Pull request by vcalasansh**
## Describe your changes
Final code changes for making Barrier Steps work within looping strategies. All behavior changes are behind the Feature Flag `CDS_NG_BARRIER_STEPS_WITHIN_LOOPING_STRATEGIES`.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52659)

### CODE REVIEW

**Comment by vcalasansh** [comment]
trigger uts

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 5 of 11 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52659)* status: 11 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/observers/BarrierExpandRequest.java` line 26 at r1](https://reviewable.io/reviews/harness/harness-core/52659#-NdxsdsTApNKGQECRRNi:-NdxsdsTApNKGQECRRNj:b-35lnwj) ([raw file](https://github.com/harness/harness-core/blob/3e9ef3d1a8da28b634b26e7b13eb833a17e776f1/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/observers/BarrierExpandRequest.java#L26)):*
> ```Java
>   List<String> childrenSetupIds;
>   List<String> childrenRuntimeIds;
>   Ambiance ambiance;
> ```

instead of passing ambiance, can we pass what is required by the apis?
___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/event/handlers/SpawnChildrenRequestProcessor.java` line 229 at r1](https://reviewable.io/reviews/harness/harness-core/52659#-NdxsqOS0i2VwdAuMr0z:-NdxsqOS0i2VwdAuMr1-:b-tspx8c) ([raw file](https://github.com/harness/harness-core/blob/3e9ef3d1a8da28b634b26e7b13eb833a17e776f1/pipeline-service/modules/orchestration/src/main/java/io/harness/event/handlers/SpawnChildrenRequestProcessor.java#L229)):*
> ```Java
>             .childrenRuntimeIds(childrenRuntimeIds)
>             .ambiance(ambiance)
>             .maxConcurrency(maxConcurrency)
> ```

how is maxConcurreny used? defining number of execution instances?
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/service/BarrierServiceImpl.java` line 520 at r1](https://reviewable.io/reviews/harness/harness-core/52659#-NdxtDtDCrnaQHKK7QOd:-NdxtDtDCrnaQHKK7QOe:b-7barwc) ([raw file](https://github.com/harness/harness-core/blob/3e9ef3d1a8da28b634b26e7b13eb833a17e776f1/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/service/BarrierServiceImpl.java#L520)):*
> ```Java
>   }
> 
>   public void upsertBarrierExecutionInstance(PlanCreationContext ctx, BarrierStepNode field) {
> ```

can we not include PlanCreationContext in service Impl, and maybe create intermediate dto and caller convert to dto for things needed by serviceImpl



_Code quote:_
```Java
PlanCreationContext ctx
```
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/service/BarrierServiceImpl.java` line 548 at r1](https://reviewable.io/reviews/harness/harness-core/52659#-NdxtS-j1uZY_h96sArB:-NdxtS-j1uZY_h96sArC:b-fepfq6) ([raw file](https://github.com/harness/harness-core/blob/3e9ef3d1a8da28b634b26e7b13eb833a17e776f1/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/service/BarrierServiceImpl.java#L548)):*
> ```Java
>       } else if (strategyId.equals(stepGroupId)) {
>         strategyNodeType = StrategyNodeType.STEP_GROUP;
>       } else {
> ```

should this logic be known to barriers? or coming from a common util generally?



_Code quote:_
```Java
      if (strategyId.equals(stageId)) {
        strategyNodeType = StrategyNodeType.STAGE;
      } else if (strategyId.equals(stepGroupId)) {
        strategyNodeType = StrategyNodeType.STEP_GROUP;
      } else {
```

**Review by archit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52659)* status: 11 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/event/handlers/SpawnChildrenRequestProcessor.java` line 86 at r1](https://reviewable.io/reviews/harness/harness-core/52659#-Ne1eRsD57vk3oafXNPk:-Ne1eRsD57vk3oafXNPl:bfnds6i) ([raw file](https://github.com/harness/harness-core/blob/3e9ef3d1a8da28b634b26e7b13eb833a17e776f1/pipeline-service/modules/orchestration/src/main/java/io/harness/event/handlers/SpawnChildrenRequestProcessor.java#L86)):*
> ```Java
>       int maxConcurrency = getMaxConcurrencyLimit(ambiance, childrenIds, request.getChildren().getMaxConcurrency());
>       if (pmsFeatureFlagService.isEnabled(
>               AmbianceUtils.getAccountId(ambiance), FeatureName.CDS_NG_BARRIER_STEPS_WITHIN_LOOPING_STRATEGIES)) {
> ```

We have not put used this FF in BarrierVisitor where we are creating barrierExecutionInstances right?
Also we should use Ambiance ff map -> as already running executions will get impacted because of this.

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/event/BarrierWithinStrategyExpander.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  @Inject private BarrierService barrierService;
+  @Inject private PersistentLocker persistentLocker;
+
+  @Override
+  public void onInitializeRequest(BarrierExpandRequest barrierExpandRequest) {
+    Ambiance ambiance = barrierExpandRequest.getAmbiance();
+    String planExecutionId = ambiance.getPlanExecutionId();
+    String strategySetupId = barrierExpandRequest.getStrategySetupId();
+    try {
+      try (AcquiredLock<?> ignore =
```
Comment (hextechpal) [inline pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/event/BarrierWithinStrategyExpander.java:49]: Move acquire lock after checking if the barriers is present in strategy

**Review by hextechpal — CHANGES_REQUESTED** [review]

**Review by vcalasansh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52659)* status: 237 file(s) to review. 2 change requests. 6. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): sto. 

___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/observers/BarrierExpandRequest.java` line 26 at r1](https://reviewable.io/reviews/harness/harness-core/52659#-NdxsdsTApNKGQECRRNi:-Ne4zbYX7MT0ixQLVC3J:b-896fja) ([raw file](https://github.com/harness/harness-core/blob/3e9ef3d1a8da28b634b26e7b13eb833a17e776f1/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/observers/BarrierExpandRequest.java#L26)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

instead of passing ambiance, can we pass what is required by the apis?
</blockquote></details>

Done!
___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/event/handlers/SpawnChildrenRequestProcessor.java` line 86 at r1](https://reviewable.io/reviews/harness/harness-core/52659#-Ne1eRsD57vk3oafXNPk:-Ne4zdhL6h2jeKM63Bc2:bxcnc3o) ([raw file](https://github.com/harness/harness-core/blob/3e9ef3d1a8da28b634b26e7b13eb833a17e776f1/pipeline-service/modules/orchestration/src/main/java/io/harness/event/handlers/SpawnChildrenRequestProcessor.java#L86)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

We have not put used this FF in BarrierVisitor where we are creating barrierExecutionInstances right?
Also we should use Ambiance ff map -> as already running executions will get impacted because of this.
</blockquote></details>

Done. Moved to use AmbianceUtils.checkIfFeatureFlagEnabled in every part which runs during pipeline execution flow.
___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/event/handlers/SpawnChildrenRequestProcessor.java` line 229 at r1](https://reviewable.io/reviews/harness/harness-core/52659#-NdxsqOS0i2VwdAuMr0z:-Ne4zmcGABbdU_z1TWEE:bgeyzjo) ([raw file](https://github.com/harness/harness-core/blob/3e9ef3d1a8da28b634b26e7b13eb833a17e776f1/pipeline-service/modules/orchestration/src/main/java/io/harness/event/handlers/SpawnChildrenRequestProcessor.java#L229)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

how is maxConcurreny used? defining number of execution instances?
</blockquote></details>

This is because of the maxConcurrency limitation: if strategy is using maxConcurrency (say it is `n`), only the first `n` barrier steps spawned within the strategy will be in sync (we only expand the BarrierPositions for the first `n` barrier steps).
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/event/BarrierWithinStrategyExpander.java` line 49 at r1](https://reviewable.io/reviews/harness/harness-core/52659#gh-1321502521:-Ne50JrN81WKldtRQIo-:b-896fja) ([raw file](https://github.com/harness/harness-core/blob/3e9ef3d1a8da28b634b26e7b13eb833a17e776f1/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/event/BarrierWithinStrategyExpander.java#L49)):*
<details><summary><i>Previously, hextechpal (Prashant Pal) wrote…</i></summary><blockquote>

Move acquire lock after checking if the barriers is present in strategy
</blockquote></details>

Done!
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/service/BarrierServiceImpl.java` line 520 at r1](https://reviewable.io/reviews/harness/harness-core/52659#-NdxtDtDCrnaQHKK7QOd:-Ne50LUD7e9nysoxd6Yi:bfo207j) ([raw file](https://github.com/harness/harness-core/blob/3e9ef3d1a8da28b634b26e7b13eb833a17e776f1/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/service/BarrierServiceImpl.java#L520)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

can we not include PlanCreationContext in service Impl, and maybe create intermediate dto and caller convert to dto for things needed by serviceImpl
</blockquote></details>

Done by making `BarrierStepPlanCreator` pass arguments explicitly.
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/service/BarrierServiceImpl.java` line 548 at r1](https://reviewable.io/reviews/harness/harness-core/52659#-NdxtS-j1uZY_h96sArB:-Ne51OCm7BlqZJnNeh4n:b9gykl3) ([raw file](https://github.com/harness/harness-core/blob/3e9ef3d1a8da28b634b26e7b13eb833a17e776f1/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/service/BarrierServiceImpl.java#L548)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

should this logic be known to barriers? or coming from a common util generally?
</blockquote></details>

I've changed it slightly. Now we are using `BarrierPositionType` instead of `StrategyNodeType`. I believe we can keep this specific to Barriers for now.

**Comment by vcalasansh** [comment]
trigger uts

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 13 of 226 files at r3, 9 of 15 files at r4, 6 of 6 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52659)* status: 239 file(s) to review. 2 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): sto. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/barrier/BarrierStepPlanCreator.java` line 44 at r5](https://reviewable.io/reviews/harness/harness-core/52659#-Ne6SjAO9mlC9Jj6Fc27:-Ne6SjAO9mlC9Jj6Fc28:b2xfhjw) ([raw file](https://github.com/harness/harness-core/blob/b5e1c3763b0c8bb62e8ba1739f80f732f0e49ad7/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/barrier/BarrierStepPlanCreator.java#L44)):*
> ```Java
>     if (featureFlagService.isEnabled(
>             ctx.getAccountIdentifier(), FeatureName.CDS_NG_BARRIER_STEPS_WITHIN_LOOPING_STRATEGIES)) {
>       String planId = ctx.getMetadata().getMetadata().getExecutionUuid();
> ```

Do you need planId or planExecutionId? if planExecutionId -> its already available in ambiance without going inside execution metadata
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/event/BarrierWithinStrategyExpander.java` line 109 at r5](https://reviewable.io/reviews/harness/harness-core/52659#-Ne6Tnqz2jomIRxMK_Ku:-Ne6Tnqz2jomIRxMK_Kv:bo8pxrq) ([raw file](https://github.com/harness/harness-core/blob/b5e1c3763b0c8bb62e8ba1739f80f732f0e49ad7/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/event/BarrierWithinStrategyExpander.java#L109)):*
> ```Java
>        after the first `maxConcurrency` steps are spawn (Barrier would be waiting for still non-spawn steps to be
>        achieved before proceeding). */
>     for (int childIndex = 0; childIndex < min(childrenSetupIds.size(), maxConcurrency); childIndex++) {
> ```

can we add condition, min cannot be zero?



_Code quote:_
```Java
min(childrenSetupIds.size(), maxConcurrency)
```
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/event/BarrierWithinStrategyExpander.java` line 111 at r5](https://reviewable.io/reviews/harness/harness-core/52659#-Ne6TwJF7nqo6CFSe4b2:-Ne6TwJF7nqo6CFSe4b3:b-p1j4jr) ([raw file](https://github.com/harness/harness-core/blob/b5e1c3763b0c8bb62e8ba1739f80f732f0e49ad7/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/event/BarrierWithinStrategyExpander.java#L111)):*
> ```Java
>     for (int childIndex = 0; childIndex < min(childrenSetupIds.size(), maxConcurrency); childIndex++) {
>       BarrierPositionBuilder newBarrierPositionBuilder = positionInfo.toBuilder().stageRuntimeId(stageExecutionId);
>       if (BarrierPositionType.STAGE.equals(positionInfo.getStrategyNodeType())) {
> ```

can we use switch case?

**Comment by sahilhindwani-harness** [comment]
trigger uts

**Review by archit-harness — COMMENTED** [review]
> Reviewed 210 of 226 files at r3, 8 of 8 files at r6, 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52659)* status: 239 file(s) to review. 2 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): sto. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/barrier/BarrierStepPlanCreator.java` line 44 at r5](https://reviewable.io/reviews/harness/harness-core/52659#-Ne6SjAO9mlC9Jj6Fc27:-Nequ8E-ExEyqpbs5avp:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/b5e1c3763b0c8bb62e8ba1739f80f732f0e49ad7/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/barrier/BarrierStepPlanCreator.java#L44)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

Do you need planId or planExecutionId? if planExecutionId -> its already available in ambiance without going inside execution metadata
</blockquote></details>

done
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/event/BarrierWithinStrategyExpander.java` line 111 at r5](https://reviewable.io/reviews/harness/harness-core/52659#-Ne6TwJF7nqo6CFSe4b2:-NequEcj-KIQwahh-Wfk:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/b5e1c3763b0c8bb62e8ba1739f80f732f0e49ad7/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/event/BarrierWithinStrategyExpander.java#L111)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

can we use switch case?
</blockquote></details>

done

**Comment by vcalasansh** [comment]
trigger uts

**Review by hextechpal — APPROVED** [review]
> Reviewed 4 of 11 files at r1, 1 of 15 files at r4, 4 of 6 files at r5, 1 of 8 files at r6, 233 of 234 files at r8, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52659)* status: 234 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained. Waiting on team(s): sto.

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 15 of 234 files at r8, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52659)* status: 241 file(s) to review. 2 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): sto. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/barrier/BarrierStepPlanCreator.java` line 44 at r9](https://reviewable.io/reviews/harness/harness-core/52659#-NfuZDFK-fZUkAN6GbUZ:-NfuZDFLAolW7pXvr17b:bcp6wea) ([raw file](https://github.com/harness/harness-core/blob/4bba900f39b1cd01f3c56a2977891830b57416da/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/barrier/BarrierStepPlanCreator.java#L44)):*
> ```Java
>     if (featureFlagService.isEnabled(
>             ctx.getAccountIdentifier(), FeatureName.CDS_NG_BARRIER_STEPS_WITHIN_LOOPING_STRATEGIES)) {
>       String planId = ctx.getMetadata().getMetadata().getExecutionUuid();
> ```

nit: its planExecutionId and not planId
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/event/BarrierWithinStrategyExpander.java` line 109 at r5](https://reviewable.io/reviews/harness/harness-core/52659#-Ne6Tnqz2jomIRxMK_Ku:-NfuZQ861HpsyLgrwmuo:b-2ayb5k) ([raw file](https://github.com/harness/harness-core/blob/b5e1c3763b0c8bb62e8ba1739f80f732f0e49ad7/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/event/BarrierWithinStrategyExpander.java#L109)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

can we add condition, min cannot be zero?
</blockquote></details>

@vcalasansh can you confirm this

**Review by vcalasansh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52659)* status: 218 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained. Waiting on team(s): sto. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/event/BarrierWithinStrategyExpander.java` line 109 at r5](https://reviewable.io/reviews/harness/harness-core/52659#-Ne6Tnqz2jomIRxMK_Ku:-Nfuv23-0oC_rPMiSZ7X:bol8al7) ([raw file](https://github.com/harness/harness-core/blob/b5e1c3763b0c8bb62e8ba1739f80f732f0e49ad7/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/barriers/event/BarrierWithinStrategyExpander.java#L109)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

@vcalasansh can you confirm this
</blockquote></details>

Thanks for the comment! I think we should be ok here, if by any chance childrenSetupIds.size() or maxConcurrency are zero, this means that we will just delete the dummy barrrierPosition for the strategy, and this should be the expected behavior. Do you see any issues with this?

**Review by vcalasansh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52659)* status: 221 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): sto. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/barrier/BarrierStepPlanCreator.java` line 44 at r9](https://reviewable.io/reviews/harness/harness-core/52659#-NfuZDFK-fZUkAN6GbUZ:-NfuxAQn449eZfdmJ4Lf:bol3kou) ([raw file](https://github.com/harness/harness-core/blob/4bba900f39b1cd01f3c56a2977891830b57416da/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/barrier/BarrierStepPlanCreator.java#L44)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

nit: its planExecutionId and not planId
</blockquote></details>

Done! Thank you!

**Comment by vcalasansh** [comment]
trigger uts

**Comment by vcalasansh** [comment]
trigger uts

**Review by archit-harness — APPROVED** [review]
> Reviewed 8 of 234 files at r8, 3 of 3 files at r10, 2 of 2 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52659)* status: 5 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): sto.

**Review by archit-harness — APPROVED** [review]

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 1 of 11 files at r1, 5 of 228 files at r3, 2 of 15 files at r4, 6 of 6 files at r5, 6 of 8 files at r6, 230 of 234 files at r8, 1 of 1 files at r9, 3 of 3 files at r10, 2 of 2 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52659)* status: Waiting on team(s): sto.

**Comment by vcalasansh** [comment]
trigger review

**Pull request merged**